### PR TITLE
Add vnodes to components part 1

### PR DESCRIPTION
**Pull request by affordances**
Also updates icons to class components.

After some tests, seems fine to do `m.Vnode<IconComponentAttrs>` instead of `m.VnodeDOM<IconComponentAttrs, this>`. Most components don't need `dom` and typing of `this` is unaffected.

### CODE REVIEW

File: packages/commonwealth/client/scripts/views/components/component_kit/cw_icons/cw_custom_icon.tsx [hunk 1]
```diff
@@ -9,12 +9,12 @@ import { CustomIconAttrs } from './types';
...
 
 export class CWCustomIcon implements m.ClassComponent<CustomIconAttrs> {
-  view(vnode) {
+  view(vnode: m.Vnode<CustomIconAttrs>) {
     const {
       componentType = ComponentType.CustomIcon,
       iconName,
       iconSize = 'medium',
-      ...domAttrs
+      ...otherAttrs
```
Comment (zakhap) [inline packages/commonwealth/client/scripts/views/components/component_kit/cw_icons/cw_custom_icon.tsx:17]: why did this change?

Comment (affordances) [reply packages/commonwealth/client/scripts/views/components/component_kit/cw_icons/cw_custom_icon.tsx:17]: `dom` has too much baggage as a word here. I just want to convey that overloads might get passed through to the DOM.

Comment (affordances) [reply packages/commonwealth/client/scripts/views/components/component_kit/cw_icons/cw_custom_icon.tsx:17]: Maybe it doesn't matter. 🤷

Comment (zakhap) [reply packages/commonwealth/client/scripts/views/components/component_kit/cw_icons/cw_custom_icon.tsx:17]: all good

**Review by zakhap — APPROVED** [review]

**Pull request merged**